microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/java/hover/PropertiesHoverParticipant.java

```diff
@@ -158,7 +159,7 @@ public Hover collectHover(JavaHoverContext context, IProgressMonitor monitor) th
 		}
 
 		// property references may be surrounded by curly braces in Java file
-		propertyKey = propertyKey.replace("{", "").replace("}", "");
+		propertyKey = propertyKey.replace("{", "").replace("}", "").replace("$", "");

```

Comment:
In https://github.com/redhat-developer/quarkus-ls/issues/425 , we discussed that this isn't the right approach because it applies to every annotation property. Certain properties like `cron` support `{..}` and `${..}` but others like `name` from `@ConfigProperty` don't and will produce unwanted behaviour.

@angelozerr has a comment, https://github.com/redhat-developer/quarkus-ls/issues/425#issuecomment-961177573 , that explains how to go about fixing this.

Reply:
Ah gotcha, so separate them